doc/userdoc/contribute/nest_with_ides.rst

```diff
@@ -0,0 +1,108 @@
+Developing NEST with IDEs
+=========================
+
+Integrated development environments can make coding more efficient and fun.
+Here are some recipes based on practical experience on how to use
+IDEs in NEST development. The information is not meant to be complete,
+cross-checked or up-to-date but is provided on a best-effort basis to get
+you started. Kindly contribute your experiences.
+
+Since we generally recommend out-of-source builds, only such are discussed here.
+
+.. contents:: We currently have recipes for the following IDEs
+   :local:
+   :depth: 2
+
+Eclipse
+-------
+
+This recipe describes how to set up Eclipse for editing, building and
+running NEST. The description here was tested on macOS 11.
```

Comment:
Maybe add that it is also tested on Ubuntu 16.04?

Reply:
Done.